src/main/scala/pirate/Read.scala

```diff
@@ -196,6 +196,7 @@ case object ReadErrorEmpty extends ReadError
 case object ReadErrorNotEnoughInput extends ReadError
 case object ReadErrorTooMuchInput extends ReadError
 case class ShowHelpText(sub: Option[String]) extends ReadError
+case class ShowVersion(version: String) extends ReadError
```

Comment:
I am not a big fan of that either but I don't think there is another way to do it at the moment given the overall structure of things.